lib/storage/src/content_manager/consensus_manager.rs

```diff
@@ -386,11 +387,30 @@ impl<C: CollectionContainer> ConsensusManager<C> {
         for single_change in &change.changes {
             match single_change.change_type() {
                 ConfChangeType::AddNode => {
-                    debug_assert!(
-                        self.peer_address_by_id()
-                            .contains_key(&single_change.node_id),
-                        "Peer should be already known"
-                    )
+                    let context = entry.get_context();
+
+                    if !context.is_empty() {
+                        let peer_uri = str::from_utf8(context)
+                            .map_err(|err| {
+                                StorageError::service_error(format!(
+                                    "failed to parse peer URI: {err}"
+                                ))
+                            })?
+                            .parse()
+                            .map_err(|err| {
+                                StorageError::service_error(format!(
+                                    "failed to parse peer URI: {err}"
+                                ))
+                            })?;
+
+                        self.add_peer(single_change.node_id, peer_uri)?;
+                    } else {
+                        debug_assert!(
+                            self.peer_address_by_id()
+                                .contains_key(&single_change.node_id),
+                            "Peer should be already known"
+                        )
+                    }
```

Comment:
Could you elaborate on why this change is needed?

Am I right that it allows us to add the very first peer, because it needs to take this path, while the others don't? And why can context be empty?

Reply:
(Discussed outside GH, posting comment here for the context.)

Yes, this is required for origin peer.

When we add other peers (i.e., other than the origin peer) to the cluster, we use `AddLearnerNode`, which carries peer URI. Once new peer synchronized with consensus, we use `AddNode` (see `try_promote_learner`) without a URI (because peer URI was already added to consensus by initial `AddLearnerNode`) to finalize the process.

I haven't tested this, but I'm almost sure that `AddLernerNode` workflow won't work for origin peer, so I extended `AddNode` handling, so that it also contains an optional URI.